### ISSUE TITLE
Rule proposal: Enforce compare function style for `Array#sort()`

### ISSUE THREAD

**Issue by fisker** [sindresorhus/eslint-plugin-unicorn#1473]
I've seen many people just write the compare function too complicated, yes sometimes maybe it's more readable, but sometimes it's just too long.

## Fail

```js
array.sort((a, b) => {
	if (a > b) {
		return 1;
	}

	if (a < b) {
		return -1;
	}

	return 0;
});
```

```js
array.sort((a, b) => a > b ? 1 : -1);
```

```js
array.sort((a, b) => {
	if (a.foo > b.foo) {
		return 1;
	}

	if (a.foo < b.foo) {
		return -1;
	}

	if (a.bar > b.bar) {
		return 1;
	}

	if (a.bar < b.bar) {
		return -1;
	}

	return 0;
});
```

```js
array.sort(() => Math.random() > 0. 5 ? 1 : -1);
```

## Pass

```js
array.sort((a, b) => a - b);
```
```js
array.sort((a, b) => a.localeCompare(b));
```

```js
array.sort((a, b) => a.foo - b.foo || a.bar - b.bar);
```

```js
array.sort(() => Math.random() - 0.5);
```

This rule should not fixable unless we are sure they are numbers.

**Comment by sindresorhus** [comment]
This is now accepted.

**Comment by papb** [comment]
> but sometimes it's just too long

Your second and fourth examples aren't long. Also, I personally prefer them, rather than the respective "pass" examples you gave.
